ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -910,10 +912,14 @@ - (void)showViewController:(ORKStepViewController *)viewController goForward:(BO
     
     NSString *progressLabel = nil;
     if ([self shouldDisplayProgressLabel]) {
-        ORKTaskProgress progress = [_task progressOfCurrentStep:viewController.step withResult:[self result]];
-        
-        if (progress.total > 0) {
-            progressLabel = [NSString stringWithFormat:ORKLocalizedString(@"STEP_PROGRESS_FORMAT", nil) ,ORKLocalizedStringFromNumber(@(progress.current + 1)), ORKLocalizedStringFromNumber(@(progress.total))];
+        if (viewController.parentReviewStep && viewController.parentReviewStep.isStandalone) {
+            progressLabel = viewController.parentReviewStep.title;
```

Comment:
Just tried that out. It's trickier than I thought. Thinks work well until you navigate back to the review step: `showViewController:` builds the progress label before changing the `currentStepViewController`. The progress label only gets build when `shouldDisplayProgressLabel` is true. But `shouldDisplayProgressLabel` returns `NO` since the (old) `currentStepViewController`'s `hideProgressLabel` property is set to `YES`. Can we safely build the progress label after setting the `currentStepViewController`?